File: API/_archive/Webhooks/golang/01_inbuilt_basic.go
// BASIC EXAMPLE: Go Inbuilt net/http
// Demonstrates native Go handling of Webhook payloads and HMAC verification.
package main

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/hex"
	"io"
	"net/http"
)

var secret = []byte("my_super_secret_webhook_key")

func main() {
	http.HandleFunc("POST /webhook", func(w http.ResponseWriter, r *http.Request) {
		payload, err := io.ReadAll(r.Body)
		if err != nil {
			http.Error(w, "Bad Request", http.StatusBadRequest)
			return
		}
		
		sig := r.Header.Get("X-Signature")

		mac := hmac.New(sha256.New, secret)
		mac.Write(payload)
		expected := hex.EncodeToString(mac.Sum(nil))

		if hmac.Equal([]byte(expected), []byte(sig)) {
			w.WriteHeader(http.StatusOK)
		} else {
			w.WriteHeader(http.StatusUnauthorized)
		}
	})
	
	http.ListenAndServe(":8080", nil)
}

File: API/_archive/Webhooks/python/01_inbuilt_basic.py
# BASIC EXAMPLE: Python Inbuilt http.server
# Demonstrates raw HTTP handling of Webhooks and HMAC SHA256 Signature verification.
import hmac
import hashlib
from http.server import BaseHTTPRequestHandler, HTTPServer

SECRET = b"my_super_secret_webhook_key"

class WebhookHandler(BaseHTTPRequestHandler):
    def do_POST(self):
        length = int(self.headers.get('content-length', 0))
        payload = self.rfile.read(length)
        signature = self.headers.get("X-Signature", "")

        # Verify Signature
        expected_sig = hmac.new(SECRET, payload, hashlib.sha256).hexdigest()
        
        if hmac.compare_digest(expected_sig, signature):
            self.send_response(200)
            print("Webhook Verified and Accepted!")
        else:
            self.send_response(401)
            print("Webhook Signature Invalid!")
            
        self.end_headers()

if __name__ == "__main__":
    print("Listening for webhooks on 8080...")
    HTTPServer(('', 8080), WebhookHandler).serve_forever()
